Documentation/README.md

```diff
@@ -57,3 +57,147 @@ tar -xf <tarball-destination> -C <path-to-extraction-root>
+
+```console
+git submodule --init --recursive
+```
+
+This will initialize the submodules and clone the repos under the src folder if they haven't been initialized. Note that this will update every submodule to their tracking commit checked in to the `source-build` repo . However you can go to each repo indivually, add custom remotes and checkout certain branches containing your changes.
+
+### Example .NET Runtime build from source
+Let's say that you want to produce a runtime that has a new API that it's implementation lives in coreclr and you want to create a .NET Core console app that uses it to test it without waiting for a new SDK or Runtime to be produced by official builds. Imagine we will add an API that just throws in Hashtable. 
+
+Hashtable implementation lives in coreclr and it's APIs are exposed in corefx in the System.Runtime.Extensions.cs contract. So in my local coreclr repo I will just add Hashtable.cs the following API:
+
+```cs
+public static void PNSE()
+{
+    throw new PlatformNotSupportedException();
+}
+```
+
+Then in my local corefx repo let's expose it in System.Runtime.Extensions.cs in the Hashtable class:
+
+```cs
+public static void PNSE() { }
+```
+
+Of course we need to commit this changes in our current repos. Once we've commited this locally, let's move to the source-build repo.
+
+Once in the source-build in our commmand prompt or terminal, lets navigate to `src/coreclr`. Once there, we will need to add a remote to the local coreclr repo, in example:
+```console
+git remote add local C:\repos\coreclr
+```
+
+Then, let's fetch the changes from our repo, by running:
+```console
+git fetch local
+```
+
+Once we've fetch the changes, we just need to checkout the branch where we made the changes:
+```console
+git checkout branchWithChanges
+```
+
+Just repeat this steps under `src/corefx` as well, to pull the corefx changes. Once we've done that, let's navigate to the `source-build` root and run:
+
+```console
+build.{cmd|sh} /p:RootRepo=core-setup
+```
+
+> The build detects you've changed the coreclr and corefx commits and offers to move them back to the tracked commits. Press `n` when you see the following message to decline and keep your changes:
+>
+> ```console
+> WARNING: submodule src/corefx, currently at 831264e53e5b9333850baa659af8a2857a9cb9b7, has diverged from checked-in
+> version 5b7674e4ae5cc782e99f50b2919dfdeb29106a46
+> If you are changing a submodule branch or moving a submodule backwards, this is expected.
+> Should I checkout src/corefx to the expected commit 5b7674e4ae5cc782e99f50b2919dfdeb29106a46 [N]o / [y]es / [q]uit
+> ```
+>
+> To skip this check and avoid pressing `n` for each build, you can either:
+>  * Set the environment variable `SOURCE_BUILD_SKIP_SUBMODULE_CHECK=1`.
+>  * Stage the submodule changes with `git add src/corefx` and/or `git add src/coreclr`.
```

Comment:
Interesting, I only put this in a quote to distinguish it as the suggestion, but I like the effect. I think I'd only have the `To skip this check` part in a quote, though--matches that kind of thing in other docs where the "normal" flow (pressing `n`) is in the main body and there's a note about how to improve it.

Reply:
Yes I was going to put this normal with no quote but I also liked the effect so I left the quote because it made it more clear. Just to make sure I understood correctly, you would have this whole section as 2 different quotes? As:

> The build detects you've changed the coreclr and corefx commits and offers to move them back to the tracked commits. Press `n` when you see the following message to decline and keep your changes:
>
> ```console
> WARNING: submodule src/corefx, currently at 831264e53e5b9333850baa659af8a2857a9cb9b7, has diverged from checked-in
> version 5b7674e4ae5cc782e99f50b2919dfdeb29106a46
> If you are changing a submodule branch or moving a submodule backwards, this is expected.
> Should I checkout src/corefx to the expected commit 5b7674e4ae5cc782e99f50b2919dfdeb29106a46 [N]o / [y]es / [q]uit
> ```

> To skip this check and avoid pressing `n` for each build, you can either:
>  * Set the environment variable `SOURCE_BUILD_SKIP_SUBMODULE_CHECK=1`.
>  * Stage the submodule changes with `git add src/corefx` and/or `git add src/coreclr`.

Or the to skip this check with no quote at all? Like:

> The build detects you've changed the coreclr and corefx commits and offers to move them back to the tracked commits. Press `n` when you see the following message to decline and keep your changes:
>
> ```console
> WARNING: submodule src/corefx, currently at 831264e53e5b9333850baa659af8a2857a9cb9b7, has diverged from checked-in
> version 5b7674e4ae5cc782e99f50b2919dfdeb29106a46
> If you are changing a submodule branch or moving a submodule backwards, this is expected.
> Should I checkout src/corefx to the expected commit 5b7674e4ae5cc782e99f50b2919dfdeb29106a46 [N]o / [y]es / [q]uit
> ```

To skip this check and avoid pressing `n` for each build, you can either:
  * Set the environment variable `SOURCE_BUILD_SKIP_SUBMODULE_CHECK=1`.
  * Stage the submodule changes with `git add src/corefx` and/or `git add src/coreclr`.

Which one of this do you like better?

Reply:
The second one, but the other way around since skipping the check is the optional (but very nice for dev workflow) part.

---

The build detects you've changed the coreclr and corefx commits and offers to move them back to the tracked commits. Press `n` when you see the following message to decline and keep your changes:

```console
WARNING: submodule src/corefx, currently at 831264e53e5b9333850baa659af8a2857a9cb9b7, has diverged from checked-in
version 5b7674e4ae5cc782e99f50b2919dfdeb29106a46
If you are changing a submodule branch or moving a submodule backwards, this is expected.
Should I checkout src/corefx to the expected commit 5b7674e4ae5cc782e99f50b2919dfdeb29106a46 [N]o / [y]es / [q]uit
```

> To skip this check and avoid pressing `n` for each build, you can either:
>   * Set the environment variable `SOURCE_BUILD_SKIP_SUBMODULE_CHECK=1`.
>   * Stage the submodule changes with `git add src/corefx` and/or `git add src/coreclr`.

Reply:
That looks nice as well. Will update. Thanks